### PR TITLE
gh-2886 FederatedStore remove double caching accidental collision.

### LINKED ISSUE #2886: FederatedStore Double Cache Collision Bug
**Describe the bug**
Having FederatedStore Rest service with no propterty gaffer.cache.service.name.suffix specified and then a addGraph for a second FederatedStore with no suffix, results in a Cache collision. Both graphs are the same. (Also a suffix could have been provided and be duplicated)

In a Graph as Service situation multiple user may add additional federated graph without a suffix and all have cache collision. This could expose inner graphs with weak/no security believing it was protected by the enclosing sub-federated graph. (equally suffix could have been provided and duplicated)

```
Rest FederatedStore{

    FedStoreA{
    owningUser=userA
    graphId = "FedFoo"
    suffix = "foo"
    cache = "FederatedCachefoo"
    graphs{
        graphAWithNoSecurityMapStore{}
    }
    }

    FedStoreB{
    owningUser=userB
    graphId = "CheatingGraph"
    suffix = "foo"
    cache = "FederatedCachefoo"
    graphs{
        graphAWithNoSecurityMapStore{} <---- exposed/stolen/duplicated
    }
    }

}
```

**To Reproduce**

Create FederatedStore Rest service with no specified suffix gaffer.cache.service.name.suffix in properties.
AddGraph FederatedStore with no specified suffix gaffer.cache.service.name.suffix in properties.
Run a FedereatedOperation{GetAllGraphIds} which goes to the subgraph.
subgraph return the graphId of itself because the two FederatedStore are the same due to a cache name collision.
Additional context
This maybe expected behaviour because the stores have been misconfigured to use the same cache.
But the possibly exploit to expose the contents of another FederatedStore in the same RestService seems incorrect.

No infinite loops should occur.

**Possible Workarounds**

Optional AddGraph Hooks to mitigate against the use of not providing a suffix
FederatedStore to mitigate against cache name collision in the same JVM, which does not effect load balancing of FederatedStore.

Impact

No one this is an Alpha 4 issue.

### PR DESCRIPTION
**Pull request by GCHQDev404**
# Related Issue

- Resolve #2886

### CODE REVIEW

**Review by t92549 — DISMISSED** [review]

File: store-implementation/federated-store/src/test/java/uk/gov/gchq/gaffer/federatedstore/FederatedStoreTest.java [hunk 1]
```diff
@@ -128,7 +129,7 @@ public class FederatedStoreTest {
     private static final String MAP_ID_1 = "miniMapGraphId1";
     private static final String FED_ID_1 = "subFedGraphId1";
     private static final String INVALID_CACHE_SERVICE_CLASS_STRING = "uk.gov.gchq.invalid";
-    private static final String CACHE_SERVICE_NAME = "federatedStoreGraphs";
+    public static final String CACHE_SERVICE_NAME = getCacheNameFrom(GRAPH_ID_TEST_FEDERATED_STORE);
```
Comment (GCHQDeveloper314) [inline store-implementation/federated-store/src/test/java/uk/gov/gchq/gaffer/federatedstore/FederatedStoreTest.java:132]: Can be private

**Review by GCHQDeveloper314 — DISMISSED** [review]
Only minor comments

File: store-implementation/federated-store/src/main/java/uk/gov/gchq/gaffer/federatedstore/FederatedStoreCacheTransient.java [hunk 2]
```diff
@@ -40,10 +41,14 @@ public FederatedStoreCacheTransient() {
...
-        super(String.format("%s%s", CACHE_SERVICE_NAME_PREFIX,
+        super(getCacheNameFrom(cacheNameSuffix));
+    }
+
+    public static String getCacheNameFrom(final String cacheNameSuffix) {
+        return String.format("%s%s", CACHE_SERVICE_NAME_PREFIX,
                 nonNull(cacheNameSuffix)
-                        ? "_" + cacheNameSuffix.toLowerCase()
-                        : ""));
+                        ? "_" + cacheNameSuffix.toLowerCase(Locale.UK)
```
Comment (t92549) [inline store-implementation/federated-store/src/main/java/uk/gov/gchq/gaffer/federatedstore/FederatedStoreCacheTransient.java:50]: Is this needed?

Comment (GCHQDev404) [reply store-implementation/federated-store/src/main/java/uk/gov/gchq/gaffer/federatedstore/FederatedStoreCacheTransient.java]: changed to locale.default() to match rest of gaffer.

**Review by GCHQDeveloper314 — APPROVED** [review]

**Review by t92549 — APPROVED** [review]

**Pull request merged**